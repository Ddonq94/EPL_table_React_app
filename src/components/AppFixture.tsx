import { useEffect, useState } from "react";
import { format } from "date-fns";
import AppClub from "./AppClub";
import { Badge, Col, Row } from "antd";

function AppFixture({ testID = "AppFixture", fixture }: any) {
  const [teamA, setTeamA] = useState<any>();
  const [teamB, setTeamB] = useState<any>();

  useEffect(() => {
    if (fixture) {
      let [homeTeam, awayTeam] = Object.entries(fixture.score);
      setTeamA([...homeTeam]);
      setTeamB([...awayTeam]);
    }
  }, [fixture]);

  const formattedDate = format(fixture.dateObj, "dd/MM HH:MM");
  const formattedDateShort = format(fixture.dateObj, "HH:MM");
  const score = teamA && teamB ? `${teamA[1]} - ${teamB[1]}` : "";

  const title = fixture.isFutureFixture ? (
    <Badge data-testid={`${testID}Badge`} count={"Future"} offset={[30, 10]}>
      <h3 data-testid={`${testID}Heading`}>{formattedDate}</h3>
    </Badge>
  ) : (
    <h3 data-testid={`${testID}Heading`}>{formattedDate}</h3>
  );

  const center = (
    <span className="fixture-center">
      {fixture.isFutureFixture ? formattedDateShort : score}
    </span>
  );

  return (
    <div data-testid={testID} className="fixture-wrapper">
      {teamA && teamB && (
        <>
          {title}
          <Row className="fixture-Row" gutter={16}>
            <Col span={12}>
              <AppClub clubName={teamA[0]} justifyEnd />
            </Col>
            <Col span={2}>{center}</Col>
            <Col span={10}>
              <AppClub clubName={teamB[0]} />
            </Col>
            <Col></Col>
          </Row>
        </>
      )}
    </div>
  );
}

export default AppFixture;
