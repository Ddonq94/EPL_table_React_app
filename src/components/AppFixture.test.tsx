import React from "react";
import { cleanup, render } from "@testing-library/react";
import AppFixture from "./AppFixture";

afterEach(() => {
  cleanup();
});

test("renders component", () => {
  const TestFn = () => {
    return <AppFixture testID="AppFixture" />;
  };

  const { getByTestId } = render(<TestFn />);
  const content = getByTestId("AppFixture");
  expect(content).toBeDefined();
  expect(content).toBeInTheDocument();
  expect(content).toBeDefined();
  expect(content).toBeEnabled();
  expect(content).not.toBeEmptyDOMElement();
});
