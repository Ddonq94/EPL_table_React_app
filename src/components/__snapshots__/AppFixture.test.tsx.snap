// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fixture rendering renders AppFixture to match snapshot 1`] = `
<div
  class="fixture-wrapper"
  data-testid="AppFixture"
>
  <h3
    data-testid="AppFixtureHeading"
  >
    04/03 17:03
  </h3>
  <div
    class="ant-row fixture-Row"
    style="margin-left: -8px; margin-right: -8px;"
  >
    <div
      class="ant-col ant-col-12"
      style="padding-left: 8px; padding-right: 8px;"
    >
      <div
        class="club-wrapper justify-end "
        data-testid="AppClub"
      >
        <img
          alt="Manchester UnitedLogo"
          data-testid="AppClubImg"
          src="https://resources.premierleague.com/premierleague/badges/25/t1.png"
        />
        <span>
           
          Manchester United
        </span>
      </div>
    </div>
    <div
      class="ant-col ant-col-2"
      style="padding-left: 8px; padding-right: 8px;"
    >
      <span
        class="fixture-center"
      >
        2 - 2
      </span>
    </div>
    <div
      class="ant-col ant-col-10"
      style="padding-left: 8px; padding-right: 8px;"
    >
      <div
        class="club-wrapper  "
        data-testid="AppClub"
      >
        <img
          alt="ChelseaLogo"
          data-testid="AppClubImg"
          src="https://resources.premierleague.com/premierleague/badges/25/t8.png"
        />
        <span>
           
          Chelsea
        </span>
      </div>
    </div>
    <div
      class="ant-col"
      style="padding-left: 8px; padding-right: 8px;"
    />
  </div>
</div>
`;
